File: Cpp/Codes/Practice/LeetCodeCN/557 Reverse words in a string iii a string iii.cpp
#include <gtest\gtest.h>

using namespace std;

string reverseWords(string s) 
{
	string result;
	auto next = s.rbegin();
	auto it = next;
	for (; it != s.rend(); ++it)
	{
		if (*it == ' ')
		{
			string value(next,it);
			result = " " + value + result;
			next = ++it;
		}
	}
	
	if (it != next)
	{
		string value(next,it);
		result = value + result;
	}
	
	return result;
}

string reverseWords1(string s)
{	
	auto next = s.rbegin();
	auto it = next;
	for (; it != s.rend(); ++it)
	{
		if (*it == ' ')
		{
			std::reverse(next, it);
			next = ++it;
		}
	}

	if (it != next)
	{
		std::reverse(next, it);
	}

	return s;
}


TEST(LeetCodeCN, tReverseWords)
{
	auto result = reverseWords1("Let's take LeetCode contest");


}

File: Python/Codes/Practice/LeetCodeCN/557 Reverse words in a string iii a string iii.py

class Solution:
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        words = s.split(' ')
        for i,v in enumerate(words):
            words[i] = v[::-1]

        return ' '.join(words)


if __name__ == '__main__':
    s = Solution()
    print(s.reverseWords("Let's take LeetCode contest"));